### ISSUE TITLE
Deep copy struct from storage to memory with array

### ISSUE THREAD

**Issue by dddejan** [SRI-CSL/solidity#87]
```solidity
pragma solidity >=0.5.0;
contract A {
  struct S {
    string s;
  }
  mapping (int=>S) m;
  function f() public view {
    S memory s = m[42];
  }
}
```

This gives:

```
./solc-verify.py Issue.sol --output .
Error while running verifier, details:
Parsing ./Issue.sol.bpl
./Issue.sol.bpl(40,20): Error: right-hand side in map store with wrong type: int_arr_type (expected: int_arr_ptr)
1 type checking errors detected in ./Issue.sol.bpl
```

The fishy part of the translation is 

```
// Deep copy struct S
s#3 := s#3[s#12 := new_array#1];
s#3 := s#3[s#12 := s#3#S#4#constr(m#8[__this][42])];
```

**Comment by hajduakos** [comment]
I'm on it. Seems like it is not limited to strings, but to arrays in general. However, it is only an issue from storage to memory, the other direction works (e.g., `StructsArrays.sol`).

**Comment by hajduakos** [comment]
Fixed in c7c2357c701fa606d1c3bb8bda68007e7f9dafbe. We were trying to copy the array into the member, but that is just a pointer into the global heap of arrays. Fixed it by copying into the global heap (indexed by the pointer).
